debian/clean

```diff
@@ -1 +1 @@
-debian/libtaskomat-????*.install
+debian/taskolib-????*.install
```

Comment:
I guess this a file from Debian build system to clean the build sources/install files, correct? Does anybody has a link where to find more info about the syntax?

Reply:
**RL;DR:** While packaging the project is installed (via the normal 'ninja -C build install'), to some temporary directory and not the prefix (of course) - the package contents will later be installed into the prefix, but at packaging time we do not want that. Ninja has no way to decide which installed files should go into which package, and this is where the `*.install` files come in. It instructs the debian packaging helpers which file of the temporary installed project go into which package, and with which final path.

Longer:

In principle the debian packaging has automatisms for 'everything'. At least for 'simple' packages.

But here we want to generate 2 packages from one project, the actual library package and a development package.
The automatisms can not know which file should go into which package, so it needs help.

The syntax of the `install` files (as well as possible `clean` file etc pp) is one line for one entry, each entry a glob for the files that shall be affected and the destination. The only problem I often have is: 'Is `**` possible here, or just `*`'.

In principle this is the reference manual: https://www.debian.org/doc/debian-policy/index.html (Sections 3, 5, 6, 7, 8)
A simpler approach / introduction can be found here: https://www.debian.org/doc/manuals/maint-guide/

For the `*.install` files it is here: https://www.debian.org/doc/manuals/maint-guide/dother.en.html#install

Hmm, which to be honest does not say much.

The debian packaging is _in principle_ highly automated and easy, but it has evolved over time and it is always changing which makes writing the specs 'complicated' if you want to support multiple debian packaging versions in parallel. This is even more complicated because there are 'Standards-Version' values which correspond to the policy manual's revisions and there a 'debhelper version' values which correspond to the helper scripts that actually build the packages. You need to keep both and their changes in your mind when writing the stuff :grimacing: 

Thank god doocs dropped Ubuntu 16.04, because it's heighest debian packaging version is retired now and can in principle not be used on recent systems anymore.

Concrete:

The 'text' before the `.install` refers to entries of packages in the `control` file.
The `*.in` file is configured by Meson into a 'real' `*.install` file.

Reply:
Oh I misread the question.

Yes, the `clean` file describes files that need to be cleaned in the end, apart from the 'expected' files.
Here it is the file created by Meson.
Cleaning can be triggered by `dh_clean` command line tool (if not triggered automatically).